tests/ut/scargo_check/ut_scargo_check.py

```diff
@@ -0,0 +1,120 @@
+from pathlib import Path
+from typing import Dict
+from unittest.mock import MagicMock
+
+import pytest
+from pytest_mock import MockerFixture
+
+from scargo.commands import check
+from scargo.commands.check import scargo_check
+from scargo.config import Config
+
+CHECKERS = [
+    check.PragmaChecker,
+    check.CopyrightChecker,
+    check.TodoChecker,
+    check.ClangFormatChecker,
+    check.ClangTidyChecker,
+    check.CyclomaticChecker,
+    check.CppcheckChecker,
+]
+
+
+@pytest.fixture
+def mock_checkers(mocker: MockerFixture) -> Dict[str, MagicMock]:
+    return {
+        checker_class.check_name: mocker.patch(
+            f"{scargo_check.__module__}.{checker_class.__name__}"
+        )
+        for checker_class in CHECKERS
+    }
+
+
+@pytest.fixture
+def mock_prepare_config(mocker: MockerFixture, config: Config) -> MagicMock:
+    return mocker.patch(
+        f"{scargo_check.__module__}.prepare_config", return_value=config
+    )
+
+
+@pytest.fixture
+def mock_get_project_root(mocker: MockerFixture) -> MagicMock:
+    return mocker.patch(
+        f"{scargo_check.__module__}.get_project_root", return_value=Path()
+    )
+
+
+def test_scargo_check_single(
```

Comment:
Maybe parametrize this? This test and `test_scargo_check_all` could be part of one big paramatrized test

Reply:
It could, but I've decided it's not worth it. It's a pain to parametrize because you need a separate param for each option, so it gets unwieldy. It can certainly be done though.

Reply:
Okay, I'm fine with this bare minimum as well